.github/workflows/coverage.yaml

```diff
@@ -15,7 +20,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          ref: ${{ matrix.branch == 'main' && 'main' || '' }}
+          ref: ${{ matrix.branch == 'main' && 'main' || (github.event.pull_request && format('refs/pull/{0}/merge', github.event.pull_request.number)) || '' }}
```

Comment:
There's a choice to be made here.

There are 3 cases:
- this is a reference build so we check out `main`
- this a pull request and we need to check out _something_ -- here we're checking out the merge head between `main` and the pull request that's what `/merge` does (this is what actions/checkout would do for the `pull_request` target -- but we have to do it manually because `pull_request_target` is treated differently)  -- there's an alternative of checking out the pull request `/head`, but I think `/merge` is the right choice
- this is a push and we're using the default behavior of letting actions/checkout get the commit for us

Reply:
merge sounds right 👍🏼